### ISSUE TITLE
System.NotSupportedException: Cannot compile: Parameter ... again?

### ISSUE THREAD

**Issue by kmustang** [praeclarum/sqlite-net#703]
Hello,

I am still having the problem described in this thread that has been closed:

[System.NotSupportedException: Cannot compile: Parameter](https://github.com/praeclarum/sqlite-net/issues/552)

It looks like I'm using version 1.4.118.  I've been using the same base database structure for years, but just started having problems a couple of weeks ago.

I defined an interface to require objects going into the database to have an integer Id:
```
public interface IHasId
{
    int Id { get; set; }
}
```

Then my "base database" class has a GetElement method that is defined like so:
```
public T GetElement<T> (int id) where T : IHasId, new ()
{
    lock (locker) {
        return Table<T>().FirstOrDefault(x => ((IHasId)x).Id == id);
    }
 }
```

That FirstOrDefault statement generates the Cannot compile error. Is there a fix/workaround for this that doesn't require me making a local copy of sqlite?

Thanks,
-Karen

**Comment by schnabs** [comment]
Try upgrading to version 1.5.231

**Comment by praeclarum** [comment]
This is fixed in 1.6 (and probably 1.5). Sorry for the trouble!
